internal/adapter/sqlite/note_dao.go

```diff
@@ -519,20 +519,26 @@ func (d *NoteDAO) findRows(opts core.NoteFindOpts, selection noteSelection) (*sq
 		return nil
 	}
 
-	if !opts.Match.IsNull() {
+	if 0 < len(opts.Match) {
 		switch opts.MatchStrategy {
 		case core.MatchStrategyExact:
-			whereExprs = append(whereExprs, `n.raw_content LIKE '%' || ? || '%' ESCAPE '\'`)
-			args = append(args, escapeLikeTerm(opts.Match.String(), '\\'))
+			for _, match := range opts.Match {
+				whereExprs = append(whereExprs, `n.raw_content LIKE '%' || ? || '%' ESCAPE '\'`)
+				args = append(args, escapeLikeTerm(match, '\\'))
+			}
 		case core.MatchStrategyFts:
 			snippetCol = `snippet(fts_match.notes_fts, 2, '<zk:match>', '</zk:match>', '…', 20)`
 			joinClauses = append(joinClauses, "JOIN notes_fts fts_match ON n.id = fts_match.rowid")
 			additionalOrderTerms = append(additionalOrderTerms, `bm25(fts_match.notes_fts, 1000.0, 500.0, 1.0)`)
-			whereExprs = append(whereExprs, "fts_match.notes_fts MATCH ?")
-			args = append(args, fts5.ConvertQuery(opts.Match.String()))
+			for _, match := range opts.Match {
+				whereExprs = append(whereExprs, "fts_match.notes_fts MATCH ?")
+				args = append(args, fts5.ConvertQuery(match))
+			}
 		case core.MatchStrategyRe:
-			whereExprs = append(whereExprs, "n.raw_content REGEXP ?")
-			args = append(args, opts.Match.String())
+			for _, match := range opts.Match {
+				whereExprs = append(whereExprs, "n.raw_content REGEXP ?")
+				args = append(args, match)
+			}
```

Comment:
Assume we want to apply the behaviour to all strategies, all AND

Reply:
Yes